spec/integration/type/file_spec.rb

```diff
@@ -292,13 +292,14 @@ def get_aces_for_path_by_sid(path, sid)
             Puppet::FileSystem.symlink(link_target, link)
           end
 
-          it "should not set the executable bit on the link nor the target" do
+          it "should not set the executable bit on the link target" do
             catalog.add_resource described_class.new(:path => link, :ensure => :link, :mode => '0666', :target => link_target, :links => :manage)
```

Comment:
If `:links => :manage`, then I think it makes sense to not try to manage the mode on the symlink or target, but we should manage the owner/group of the symlink (eg https://unix.stackexchange.com/questions/224979/why-do-linux-posix-have-lchown-but-no-lchmod).

If `:links => :follow`, then I would expect puppet to change the target's mode.

Do we have tests for that?

Reply:
I'm not immediately seeing anything that looks like it's testing for ownership when symlinks are involved.